### ISSUE TITLE
donlog-normal broken

### ISSUE THREAD

**Issue by roedie** [haproxy/haproxy#1754]
### Detailed Description of the Problem

After upgrading from 2.4.17 to 2.6.0 it seems like `option dontlog-normal` is not working anymore. Everything is logged.

### Expected Behavior

Only errors are logged.

### Steps to Reproduce the Behavior

Upgrade from 2.4.17 to 2.6.0.

### What is your configuration?

```haproxy
frontend redacted
        bind a.b.c.d:80 transparent
        bind 2001:ffff::c:80 transparent

        bind a.b.c.d:443 ssl crt /path/to/cert.pem transparent alpn h2,http/1.1
        bind 2001:ffff::c:443 ssl crt /path/to/cert.pem transparent alpn h2,http/1.1

        mode http
        maxconn 8192

        option httplog
        option dontlog-normal
        option http-ignore-probes
        option forwardfor

        default_backend cluster

backend cluster
        fullconn        4096
        mode    http

        balance roundrobin
        option redispatch
        option forwardfor
        option httpchk GET /are-you-alive
        http-check expect status 200
        retries 3
        retry-on all-retryable-errors

        default-server check check-ssl check-alpn http/1.1 maxconn 500 weight 100 inter 2s rise 3 fall 3 agent-check agent-port 8081 agent-inter 20s ssl verify none alpn h2,http/1.1
        server server-11 a.b.c.d:8888
        server server-12 a.b.c.d:8888
        server server-13 a.b.c.d:8888
        server server-14 a.b.c.d:8888
        server server-15 a.b.c.d:8888
        server server-16 a.b.c.d:8888

        # Sorry Server
        server outage 127.0.0.1:80 backup no-check
```

### Output of `haproxy -vv`

```plain
HAProxy version 2.4.17-1~bpo11+1 2022/05/14 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2026.
Known bugs: http://www.haproxy.org/bugs/bugs-2.4.17.html
Running on: Linux 5.10.0-15-cloud-amd64 #1 SMP Debian 5.10.120-1 (2022-06-09) x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wdeclaration-after-statement -fwrapv -Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference
  OPTIONS = USE_PCRE2=1 USE_PCRE2_JIT=1 USE_OPENSSL=1 USE_LUA=1 USE_SLZ=1 USE_SYSTEMD=1 USE_PROMEX=1
  DEBUG   =

Feature list : +EPOLL -KQUEUE +NETFILTER -PCRE -PCRE_JIT +PCRE2 +PCRE2_JIT +POLL -PRIVATE_CACHE +THREAD -PTHREAD_PSHARED +BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H +GETADDRINFO +OPENSSL +LUA +FUTEX +ACCEPT4 -CLOSEFROM -ZLIB +SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL +SYSTEMD -OBSOLETE_LINKER +PRCTL -PROCCTL +THREAD_DUMP -EVPORTS -OT -QUIC +PROMEX -MEMORY_PROFILING

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=4).
Built with OpenSSL version : OpenSSL 1.1.1n  15 Mar 2022
Running on OpenSSL version : OpenSSL 1.1.1n  15 Mar 2022
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.3.3
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with libslz for stateless compression.
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.36 2020-12-04
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with gcc compiler version 10.2.1 20210110

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2       flags=HTX|CLEAN_ABRT|HOL_RISK|NO_UPG
            fcgi : mode=HTTP       side=BE        mux=FCGI     flags=HTX|HOL_RISK|NO_UPG
       <default> : mode=HTTP       side=FE|BE     mux=H1       flags=HTX
              h1 : mode=HTTP       side=FE|BE     mux=H1       flags=HTX|NO_UPG
       <default> : mode=TCP        side=FE|BE     mux=PASS     flags=
            none : mode=TCP        side=FE|BE     mux=PASS     flags=NO_UPG

Available services : prometheus-exporter
Available filters :
        [SPOE] spoe
        [CACHE] cache
        [FCGI] fcgi-app
        [COMP] compression
        [TRACE] trace
```

**Comment by chipitsine** [comment]
looks similar to https://github.com/haproxy/haproxy/issues/1594

**Comment by roedie** [comment]
It looks a bit the same, but since this has always worked in previous versions of HAProxy it is kind of strange. I think I haven't changed this bit since 1.5.X or so...

**Comment by capflam** [comment]
Indeed. It is a regression of the commit 731c8e6cf. I will fix it. Thanks !

**Comment by capflam** [comment]
Fixed ! It will be backported soon and shipped with the 2.6.1 (planned for the next week).

**Comment by wtarreau** [comment]
fix was backported to 2.6.1, now closing.
